### PR TITLE
librbd: add create timestamp metadata for image

### PR DESCRIPTION
**Pull request by runsisi**
Signed-off-by: runsisi <runsisi@zte.com.cn>

### CODE REVIEW

**Comment by runsisi** [comment]
retest this please

File: src/cls/rbd/cls_rbd.cc [hunk 1]
```diff
@@ -223,19 +223,23 @@ int create(cls_method_context_t hctx, bufferlist *in, bufferlist *out)
   bufferlist featuresbl;
   bufferlist object_prefixbl;
   bufferlist snap_seqbl;
+  bufferlist creation_timestampbl;
```
Comment (dillaman) [inline src/cls/rbd/cls_rbd.cc]: Nit: perhaps change ```creation_xyz``` to just ```create_xyz``` throughout

File: src/librbd/image/OpenRequest.cc [hunk 2]
```diff
@@ -322,6 +322,48 @@ Context *OpenRequest<I>::handle_v2_get_stripe_unit_count(int *result) {
...
+Context *OpenRequest<I>::handle_v2_get_creation_timestamp(int *result) {
+  CephContext *cct = m_image_ctx->cct;
+  ldout(cct, 10) << this << " " << __func__ << ": r=" << *result << dendl;
+
+  if (*result == 0) {
+    bufferlist::iterator it = m_out_bl.begin();
+    *result = cls_client::get_creation_timestamp_finish(&it,
+        &m_image_ctx->creation_timestamp);
+  }
+  if (*result == -EOPNOTSUPP) {
```
Comment (dillaman) [inline src/librbd/image/OpenRequest.cc]: Nit: you can drop this no-op clause and just update the next conditional to ```(*result < 0 && *result != -EOPNOTSUPP)```

File: src/tools/rbd/action/Info.cc [hunk 3]
```diff
@@ -187,6 +198,13 @@ static int do_show_info(librados::IoCtx &io_ctx, librbd::Image& image,
     }
   }
 
+  if (f) {
+    f->dump_string("creation_timestamp", creation_timestamp_str);
+  } else {
+    std::cout << "\tcreation_timestamp: " << creation_timestamp_str
```
Comment (dillaman) [inline src/tools/rbd/action/Info.cc]: Nit: only display if you actually have a timestamp

File: src/pybind/rbd/rbd.pyx [hunk 4]
```diff
@@ -228,6 +228,7 @@ cdef extern from "rbd/librbd.h" nogil:
                             uint8_t enabled)
     int rbd_get_stripe_unit(rbd_image_t image, uint64_t *stripe_unit)
     int rbd_get_stripe_count(rbd_image_t image, uint64_t *stripe_count)
+    int rbd_get_timestamp(rbd_image_t image, timespec *timestamp)
```
Comment (dillaman) [inline src/pybind/rbd/rbd.pyx]: Nit: delete from this commit

**Comment by runsisi** [comment]
@dillaman updated as you suggested, thanks!

**Pull request merged**